### DISCUSSION TITLE
Getting metadata about R-universe

### DISCUSSION THREAD

**Discussion by mccarthy-m-g** [r-universe-org/help#224] [Q&A] [answered]
Is it possible to get metadata about the entire R-universe programmatically? I am trying to see how many unique packages are hosted on R-universe but not on CRAN.

In particular, I would like to use `available.packages()` to get all the packages on R-universe. This can be done for individual repositories:

```r
available.packages(contrib.url("https://ropensci.r-universe.dev"))
```

But I'm not aware of an endpoint for the entire R-universe.

Alternatively, is there an easy way to get a vector of all the maintainers in R-universe? (aside from web scraping). Then I could just iterate over everyone like:

```r
lapply(
  list("https://ropensci.r-universe.dev", "https://tidyverse.r-universe.dev"),
  function(x) available.packages(contrib.url(x))
)
```

**Comment by jeroen** [comment, marked as answer]
Packages in universes may overlap, so there is no such thing as a single repository with all packages. Each one is unique.

You can get a list of all universes like this:

```r
df <- jsonlite::stream_in(url("https://r-universe.dev/stats/universes"))
df$universe
```

**Reply by mccarthy-m-g** [reply]
Thanks! Surprisingly there were no duplicate packages across universes.

In case you were curious, I found that there are 1719 packages available on R-universe but not CRAN or Bioconductor.
